framework/Util/Point.swift

```diff
@@ -1,9 +1,10 @@
 // class defining a point
 public struct Point {
-  public var x : Float
-  public var y : Float
+  public var x : Float = 0
+  public var y : Float = 0
```

Comment:
```suggestion
  public let x: Float
  public let y: Float
```

I'd make these let, because you're generally not going to want to mutate them inside of the Point. Instead, you'd perform arithmetic on the Point itself to generate a new Point.